src/pages/Top.js

```diff
@@ -9,10 +9,23 @@ const Top = () => {
   const [permission, setPermission] = useState('');
   const [accountData, setAccountData] = useState({});
 
+  const usePrevious = (value) => {
+    const ref = useRef();
+    useEffect(() => {
+      ref.current = value;
+    });
```

Comment:
Let's also add `value` to this `useEffect`'s dependency array like in this article: https://usehooks.com/usePrevious/.

<img width="692" alt="Screen Shot 2021-02-26 at 12 04 37 PM" src="https://user-images.githubusercontent.com/32081352/109349463-d6a48d80-782a-11eb-823e-64390d07fd42.png">